packages/protocol/contracts/L1/v1/V1Proposing.sol

```diff
@@ -53,13 +60,32 @@ library V1Proposing {
 
         _validateMetadata(meta);
 
-        bytes32 commitHash = _calculateCommitHash(
-            meta.beneficiary,
-            meta.txListHash
-        );
-
-        require(isCommitValid(s, commitHash), "L1:commit");
-        delete s.commits[commitHash];
+        if (LibConstants.K_COMMIT_DELAY_CONFIRMATIONS > 0) {
+            bytes32 commitHash = _calculateCommitHash(
+                meta.beneficiary,
+                meta.txListHash
+            );
+
+            require(
+                isCommitValid(
+                    s,
+                    meta.commitSlot,
+                    meta.commitHeight,
+                    commitHash
+                ),
+                "L1:notCommitted"
+            );
+
+            if (meta.commitSlot == 0) {
+                // Special handling of slot 0 for refund; non-zero slots
+                // are supposed to managed by node software for reuse.
+                // Why:
+                // - if use slot 0, a commit's gas cost is 20000 + 2900 - 4800 = 18100
+                // - if use non-zero slots, the cost is 20000 for the first time, then
+                //   2900 each time afterwards.
```

Comment:
It's been a while since I did these kind of things, but looking at some of the references again isn't the first SSTORE gas cost `22100` now? And a non-zero to non-zero SSTORE gas cost `5000` (although the hash value being read could in theory be removed, it won't have an impact on the gas cost because writing will also take the `2100` gas cost hit because it's the first access).

Reply:
I'm using info on this link https://github.com/wolflo/evm-opcodes/blob/main/gas.md#a7-sstore, is it deprecated? But I think even if there are small differences, the end results should be similiar.

Reply:
Seem up to date, though hard to follow. :) This page makes it a bit easier to see the cost for common cases: https://hackmd.io/@fvictorio/gas-costs-after-berlin

I think for some of numbers in the comments the cold access cost of `2100` isn't added, but yeah doesn't matter that much.